[FIO toup] zynqmp: use primary offset by default

igoropaniuk opened the pull request:
Restart Golden Image Search instead of hang if boot.bin was booted from unsupported multiboot offset.

Please do not submit a Pull Request via github.  Our project makes use of
mailing lists for patch submission and review.  For more details please
see https://www.denx.de/wiki/U-Boot/Patches

The only exception to this is in order to trigger a CI loop on Azure prior
to posting of patches.

MrCry0 reviewed (commented):
What issue fixes this commit?

ricardosalveti commented:
This is marked as internal, would it be something to upstream?

@caiotpereira can you test the patch as well? Thanks!

igoropaniuk commented:
@ricardosalveti fixed

MrCry0 commented:
> What issue fixes this commit?

@igoropaniuk Thanks, I've read the thread and I'd suggest adding some explanation to the commit message and maybe even to the code. About overflowing the image offset in case the boot device has size <= 32Mb.

ricardosalveti reviewed (approved):
LGTM, please merge once tested and validated (@caiotpereira )

igoropaniuk commented:
@MrCry0 technically it doesn't address only this particular case.  Even if we flash `boot.bin` to some random address but in the range of these 32 MBs, it's better to restart Golden Image Search.

igoropaniuk commented:
Testing has shown that `do_reset()` doesn't reset the board, and it turned out that SPL doesn't have a proper implementation for reset. U-Boot proper uses PSCI exposed by TF-A, which communicates with PMU FW for a proper reset (https://github.com/ARM-software/arm-trusted-firmware/blob/master/plat/xilinx/zynqmp/plat_psci.c#L161). So we have two options now:

1. Enable watchdog in U-Boot, so it reset after timeout
2. Add support for PMU API, so it can be used from SPL

MrCry0 commented:
> @MrCry0 technically it doesn't address only this particular case. Even if we flash `boot.bin` to some random address but in the range of these 32 MBs, it's better to restart Golden Image Search.

@igoropaniuk ok, thanks for an explanation. got it.

ricardosalveti commented:
> * Enable watchdog in U-Boot, so it reset after timeout
> * Add support for PMU API, so it can be used from SPL

I guess whatever is simpler :-)

But having watchdog functional in SPL is also a plus.

igoropaniuk committed:
[FIO toup] zynqmp: use primary offset by default

Use CONFIG_SYS_SPI_U_BOOT_OFFS instead of hang() if boot.bin was booted
from unsupported multiboot offset.

Fixes: cb96e38 ("[FIO toup] zynqmp: add dynamic detection of fit image spi offset")
Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

igoropaniuk commented:
@ricardosalveti I've changed it a bit, so now it force uses `CONFIG_SYS_SPI_U_BOOT_OFFS` for a FIT image for all non-B boot offsets. It doesn't introduce any security risk (IMO), as even if some unknown `boot.bin` at some random offset is loaded, it's still have to be signed with RoT key, but there is a risk that U-Boot signature verification will fail.

Let me know what you think about this approach.

ricardosalveti commented:
> @ricardosalveti I've changed it a bit, so now it force uses `CONFIG_SYS_SPI_U_BOOT_OFFS` for a FIT image for all non-B boot offsets. It doesn't introduce any security risk (IMO), as even if some unknown `boot.bin` at some random offset is loaded, it's still have to be signed with RoT key, but there is a risk that U-Boot signature verification will fail.
> 
> Let me know what you think about this approach.

Looks fine, was it tested and validated by @caiotpereira ? If so, feel free to merge.
